compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/AnnotationArgumentReference.kt

```diff
@@ -263,11 +265,19 @@ public sealed class AnnotationArgumentReference {
           is KtCollectionLiteralExpression ->
             psiElement.children.map { parsePsiElement(it) }.convertToArrayIfNeeded()
           is KtConstantExpression, is KtPrefixExpression -> parsePrimitiveType(psiElement.text)
-          is KtStringTemplateExpression ->
-            psiElement
-              .getChildOfType<KtStringTemplateEntry>()
-              ?.text
-              ?: fail()
+          is KtStringTemplateExpression -> {
+            val parts = psiElement.getChildrenOfType<KtStringTemplateEntry>()
+
+            parts.map { templateEntry ->
+              if (templateEntry is KtSimpleNameStringTemplateEntry ||
+                templateEntry is KtBlockStringTemplateEntry
+              ) {
+                parsePsiElement(templateEntry.expression as PsiElement)
```

Comment:
It seems like the bug is that we're calling getChildOfType instead of getChildrenOfType which is why the rest of the string doesn't come along. Why are you parsing both of the elements instead of just doing a joinToString(", ") on both parts?

Reply:
In this case there were two separate issues. You're correct about the first one which was that we were only grabbing text for the first part of the template regardless of template length. The second issue was that we were just treating the template as plain text, which results in constants staying as their code text instead of their value (e.g. returning `${CONSTANT}` instead of resolving it).